lib/panel-league/basic.js

```diff
@@ -73,11 +73,17 @@ const handleSwapping = ((state) => {
     } else if (block.swapTimer < 0) {
       ++block.swapTimer;
     }
+    else {
+      state.dirty = true;
+    }
   });
 });
 
 // Match three or more similar blocks horizontally or vertically
 const findMatches = ((state, blocks) => {
+  if (!state.dirty) {
```

Comment:
This function is really expensive. I'd really like to short-circuit as soon as possible with no assignment overhead.